eth/api.go

```diff
@@ -543,3 +543,42 @@ func (api *PrivateDebugAPI) getModifiedAccounts(startBlock, endBlock *types.Bloc
 	}
 	return dirty, nil
 }
+
+type AccessListResult struct {
+	Accesslist *types.AccessList `json:"accessList"`
+	Error      string            `json:"error,omitempty"`
+	GasUsed    hexutil.Uint64    `json:"gasUsed"`
+}
+
+// CreateAccessList creates a EIP-2930 type AccessList for the given transaction.
+// Its the result of the `debug_createAccessList` RPC call.
+// Reexec and BlockNrOrHash can be specified to create the accessList on top of a certain state.
+// The AccessListResult contains an error if the transaction failed.
+func (api *PublicDebugAPI) CreateAccessList(blockNrOrHash rpc.BlockNumberOrHash, reexec uint64, args *ethapi.SendTxArgs) (*AccessListResult, error) {
```

Comment:
I'd like to request to conform to the other APIs where the SendTxArgs are the first params and the block number is the **optional** second, defaulting to `latest`. We should make `reexec` optional too. Please check that it works from the RPC side too without specifying anything other than the send args.